explore/bulk-pipeline/bulk.ipynb

```diff
@@ -148,140 +175,107 @@
     "hetmat.path_counts_cache = hetmech.hetmat.caching.PathCountPriorityCache(hetmat, allocate_GB=16)\n",
     "mean_dwpcs = dict()\n",
     "for metapath in tqdm.tqdm(metapaths):\n",
-    "    row, col, dwpc_matrix = pipeline.compute_save_dwpc(hetmat, metapath, damping=0.5)\n",
+    "    row_ids, col_ids, dwpc_matrix = hetmech.degree_weight.dwpc(hetmat, metapath, damping=0.5, dense_threshold=0.7, dtype='float64')\n",
     "    mean_dwpcs[(metapath, 'dwpc', 0.5)] = dwpc_matrix.mean()\n",
+    "    path = hetmat.get_path_counts_path(metapath, 'dwpc', 0.5, None)\n",
+    "    if not path.exists():\n",
+    "        hetmech.hetmat.save_matrix(dwpc_matrix, path)\n",
     "    del dwpc_matrix"
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": 6,
+   "execution_count": 7,
    "metadata": {},
-   "outputs": [],
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "PathCountPriorityCache containing 2 items\n",
+      "  total gets: 2\n",
+      "  cache hits: memory = 0, disk = 2, absent = 0\n",
+      "  2.06 GB in use of 16.00 GB allocated\n"
+     ]
+    }
+   ],
    "source": [
+    "print(hetmat.path_counts_cache.get_stats())\n",
     "hetmat.path_counts_cache = None"
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "### Generate DGP files\n",
-    "One file per metapath per permutation"
+    "### Generate running DGP metrics"
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": 7,
+   "execution_count": 8,
    "metadata": {},
    "outputs": [
     {
      "name": "stderr",
      "output_type": "stream",
      "text": [
-      "100%|██████████| 25/25 [00:00<00:00, 278.45it/s]\n"
+      "100%|██████████| 200/200 [1:59:09<00:00, 35.75s/it]\n"
      ]
     }
    ],
    "source": [
     "for name, permat in tqdm.tqdm(hetmat.permutations.items()):\n",
     "    permat.path_counts_cache = hetmech.hetmat.caching.PathCountPriorityCache(permat, allocate_GB=16)\n",
     "    for metapath in metapaths:\n",
-    "        path = permat.directory.joinpath('degree-grouped-path-counts', 'dwpc-0.5', f'{metapath}.tsv')\n",
+    "        degree_grouped_df = hetmech.degree_group.single_permutation_degree_group(\n",
+    "            permat, metapath, dwpc_mean=mean_dwpcs[(metapath, 'dwpc', 0.5)], damping=0.5)\n",
+    "        path = hetmat.get_running_degree_group_path(metapath, 'dwpc', 0.5, extension='.pkl')\n",
     "        if path.exists():\n",
-    "            pass\n",
+    "            running_df = pandas.read_pickle(path)\n",
```

Comment:
Could we move this to a function in degree_group or pipeline so that there is some function that computes DGP summaries across permutations? This being, since we delete `summarize_degree_grouped_permutations`